components/usePublishDate.js

```diff
@@ -0,0 +1,9 @@
+import { DateTime } from 'luxon';
+
+export const usePublishDate = (page) => {
+  const day = DateTime.fromISO(page.publish_date).setLocale('en-US').day;
```

Comment:
We don't need to create the date 3 times here. We can easily create one time as `const date = DateTime.fromISO(...);` and use this date to get day for instance `date.day`